pyvista/core/composite.py

```diff
@@ -809,14 +813,47 @@ def flatten(
               .. warning::
 
                   This mode may result in duplicate key names if the same block name is
-                  reused in any nested blocks.
+                  reused in any nested blocks. Use ``safe_update=True`` to raise an
+                  error instead.
 
             - ``'prepend'``: Preserve the block names and prepend the parent names.
             - ``'reset'``: Reset the block names to default values.
 
+        field_data_mode : 'preserve' | 'prepend', default: 'preserve'
+            Mode for naming the root field data keys when moving nested field data.
+
+            - ``'preserve'``: The array names of nested field data are preserved.
+            - ``'prepend'``: Preserve the array names and prepend the parent names.
+
+        user_dict_mode : 'preserve' | 'prepend' | 'flat' | 'nested', default: 'preserve'
+            Mode for naming the flattened :attr:`~pyvista.DataObject.user_dict` keys when
+            nested :class:`MultiBlock` blocks define a user-dict.
+
+            - ``'preserve'``: Update the root user dict directly with the items of any
+              nested user-dict.
+            - ``'nested'``: Create nested keys in the root user-dict which match the
+              nested hierarchy of any nested ``MultiBlock`` blocks.
+            - ``'flat'``: Create a new key in the root user dict for each nested
+              ``MultiBlock`` that has a user-dict.
+            - ``'prepend'``: Similar to ``'flat'`` except the key names are prepended
+              with the parent block names.
+
+            .. note::
+                If there is only a single level of nesting the ``'flat'``, ``'nested'``
+                and ``'prepend'`` modes are all equivalent. They only differ when there
+                is at least two levels of nesting.
+
         separator : str, default: '::'
-            String separator to use when ``name_mode='prepend'`` is used. The separator
-            is inserted between parent and child block names.
+            String separator to use when ``'prepend'`` mode is used. The separator
+            is inserted between parent and child block name or field data array names.
+
+        safe_update : bool, default: True
```

Comment:
This name matches the keyword name from `move_nested_field_data_to_root`, but I'm not sure it's a meaningful description for `flatten`. Should this perhaps be called `check_duplicate_keys` or something similar? Should the keyword in `move_nested_field_data_to_root` also be updated to match?
```suggestion
        check_duplicate_keys : bool, default: True
```

Reply:
+1 `check_duplicate_keys`. I am always a big fan of explicit expression!